### ISSUE TITLE
Postgres 9.4: invalid input syntax for type boolean

### ISSUE THREAD

**Issue by conartist6** [sqlalchemy/sqlalchemy#5041]
Database: Postgres 9.4.24
Python: 2.7.15 64 bit
SQLAlchemy: 1.3.12

SQLAlchemy is generating the following SQL (edited for clarity) which seems to be invalid on this version of Postgres:
```sql
SELECT
	project_users.id AS project_users_id,
	project_users.user_id AS project_users_user_id
FROM
	users,
	project_users
	LEFT OUTER JOIN users AS users_1 ON users_1._id = project_users.user_id
WHERE (project_users.project, users._id = project_users.user_id)
IN(('4c472b93-ead5-ee93-b56e-478cb0bc4dc5', '5b71cfa0d2255300289e4d63'))
```

The resultant error is:
```
Query 1 ERROR: ERROR:  invalid input syntax for type boolean: "5b71cfa0d2255300289e4d63"
LINE 9: IN(('4c472b93-ead5-ee93-b56e-478cb0bc4dc5', '5b71cfa0d225530...
```

The python code looks like:
```
class ProjectUser(BaseModel):
    __tablename__ = 'project_users'

    id = Column(fields.Integer, primary_key=True, autoincrement=True)
    user_id = Column(fields.UID, ForeignKey(User._id), nullable=False)
    project = Column(fields.UID, ForeignKey(Project.uid), nullable=False)
    user = relationship("User", lazy='select')

class User(BaseModel):
    __tablename__ = 'users'

    _id = Column(fields.UID, nullable=False)

with read_only_transaction() as session:
    query = session.query(ProjectUser)
    query = query.options(joinedload(ProjectUser.user))

    project_users = query.filter(tuple_(ProjectUser.project, ProjectUser.user).in_(project_user_ids)).all()
```

What went wrong?

**Comment by conartist6** [comment]
If I change the query around it works fine:
```sql
SELECT
	project_users.id AS project_users_id,
	project_users.user_id AS project_users_user_id
FROM
	users,
	project_users
	LEFT OUTER JOIN users AS users_1 ON users_1._id = project_users.user_id
WHERE
	users._id = project_users.user_id
	AND(project_users.project, project_users.user_id)
	IN(('4c472b93-ead5-ee93-b56e-478cb0bc4dc5', '5b71cfa0d2255300289e4d63'))
```

**Comment by zzzeek** [comment]
does the error go away if you remove the joinedload ?

**Comment by zzzeek** [comment]
oh.  you're putting "Project.user", which is in a relationship, inside of a tuple.  taht's not going to do anything useful.

**Comment by conartist6** [comment]
Good question. I just tried it without the joinedLoad. I thought the problem would go away but it does not. The new SQL is
```sql
SELECT
	project_users.id AS project_users_id,
	project_users.user_id AS project_users_user_id
FROM
	project_users,
	users
WHERE (project_users.project, users._id = project_users.user_id)
IN(('4c472b93-ead5-ee93-b56e-478cb0bc4dc5', '5b71cfa0d2255300289e4d63'))
```

**Comment by zzzeek** [comment]
in fact the SQL it's generating is exactly what it would do, the SQL expression of ProjectUser.user is "users._id = project_users.user_id".  There's a special case that catches it when you do an `__eq__` that rewrites it which is likely what you're looking for but none of that is going to work inside of a tuple_() right now.   it would be an involved project to make that work.

**Comment by zzzeek** [comment]
Sorry!   I know what you were going for and it would be nice if that worked someday.

**Comment by conartist6** [comment]
Thanks for helping me catch that... It would be nice to have that support (or an error) eventually, but for now I'm unblocked and at least if anyone else makes this same typo they'll find this issue and know what they did wrong : )

**Comment by zzzeek** [comment]
the ideal is things like that would work.   but the" try it and see if it works, or explodes" is actually what motivates the name "alchemy".   so dust yourself off and try again! :)
